appsec/src/extension/commands_helpers.c

```diff
@@ -749,9 +757,33 @@ bool dd_command_process_telemetry_metrics(mpack_node_t metrics)
             if (mpack_node_error(metrics) != mpack_ok) {
                 break;
             }
+            if (dd_msgpack_helpers_is_data_truncated() &&
+                strncmp(WAF_REQUEST_TAG, key_str, WAF_REQUEST_TAG_LEN) == 0) {
```

Comment:
`strncmp` will compare _at most_ `WAF_REQUEST_TAG_LEN` characters, meaning that if `WAF_REQUEST_TAG` was a substring of `key_str`, it would consider them equally. 

You probably want something along the lines of: 
```suggestion
                WAF_REQUEST_TAG_LEN == key_len && memcmp(WAF_REQUEST_TAG, key_str, WAF_REQUEST_TAG_LEN) == 0) {
```